src/Identity/UI/src/Areas/Identity/Pages/V3/Account/ExternalLogin.cshtml.cs

```diff
@@ -204,10 +204,15 @@ public override async Task<IActionResult> OnPostConfirmationAsync(string returnU
                         await _signInManager.SignInAsync(user, isPersistent: false);
                         var userId = await _userManager.GetUserIdAsync(user);
                         var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                        while (code.Contains('+'))
```

Comment:
Should these change to do-while loops to avoid the repeated code?

Reply:
Or, perhaps more interestingly, I don't understand why this check is necessary, nor would anyone reading this code (this can end up in a user's project!), so should this "feature" be built-in to the framework?

Reply:
The issue is that we html encode the code in these pages before sending the email.  So this is working around the fact that '+' is illegal in our html encoder, I guess we could consider moving this into the identity token generation itself but that seems a bit weird, @blowdart thoughts?

Reply:
Why not have the token generator avoid characters that aren't compatible? Can it be done in a non-breaking way? (I would imagine that the character set used by the API is an implementation detail and not contractual.)

Reply:
The token generators are pluggable, so even if we fix the default one, without protection in the UI layer, its likely this + issue could still be hit by other token generators.  We could just add a helper method that takes in a func to generate the code, and the call to generate the code/send the email to at least consolidate this into one place

Reply:
If it's pluggable, how do we know that `+` is the only invalid character? What if my generator puts all kinds of `)*#^%%&*@!` in there?

Reply:
We don't, so that's what the helper method could be responsible for, ensuring we have something that's able to be html encodable before sending the email, but that's a UI concern since the html encoding is not something the framework is doing at all

Reply:
OK. Isn't everything HTML encodable though? Well, minus control characters, maybe?

Reply:
Going to back out the + fix from this PR for now since its more involved than the other minor tweaks, will come back to that in a separate PR